app/models/g5_authenticatable/user.rb

```diff
@@ -47,6 +47,23 @@ def update_roles_from_auth(auth_data)
       end
     end
 
+    def clients
+      return G5Updatable::Client.all if has_global_role
+      G5Updatable::Client.where(id: client_roles.map(&:resource_id))
+    end
+
+    def client_roles
+      G5Authenticatable::Role.joins('INNER JOIN g5_updatable_clients ON g5_updatable_clients.id = g5_authenticatable_roles.resource_id')
+        .where('g5_authenticatable_roles.resource_type = ?', G5Updatable::Client.name)
+    end
+
+    def has_global_role
+      has_role?(:admin) ||
```

Comment:
I should note that moving this to the policy would mean that the `G5Authenticatable::User#clients` method would only return scoped clients, and we would rely on the new client policy scope to know that users with global roles should be able to access all clients. So anytime you need a list of clients for the `current_user` in a controller, you'd have to do the lookup through pundit with code like this:

```ruby
@clients = policy_scope(G5Updatable::Client)
```

And then the policy scope in g5_header might look something like this (untested code which might differ from the actual implementation):

```ruby
module G5Updatable
  class ClientPolicy < ApplicationPolicy
    class Scope
      def resolve
        if has_global_role?
          scope.all
        elsif !user.client_roles.empty?
          # This might change so that we use scope.where
          # and pass in the info from user.clients as a condition
          user.clients
        else
          scope.none
        end
    end
  end
end
```

Eventually all of the apps should be running everything through pundit policies to make sure that they do not return data that the authenticated user is not authorized to access. However, it does represent a pretty significant deviation from the way that client apps currently look up their data.